helpers/utils.py

```diff
@@ -11,25 +11,41 @@
 from tempfile import NamedTemporaryFile
 from typing import Dict, List, Optional, Union
 
+import constants
 import yaml
 
-USER_HOME_DIR_ENT_IDX = 5
+PathLike = Union[str, "os.PathLike[str]"]
 
-EXIT_CODE_BAD_KUBECONFIG = -100
-EXIT_CODE_BAD_CONF_ARG = -200
-EXIT_CODE_GET_SECRET_FAILED = -300
-EXIT_CODE_PUT_SECRET_ENV_FILE_FAILED = -400
-EXIT_CODE_PUT_SECRET_LITERAL_FAILED = -500
-EXIT_CODE_DEL_SECRET_FAILED = -600
-EXIT_CODE_GET_K8S_PROPS_FAILED = -700
-EXIT_CODE_GET_PRIMARY_RESOURCES_FAILED = -800
-EXIT_CODE_SET_PRIMARY_RESOURCES_FAILED = -900
 
-PathLike = Union[str, "os.PathLike[str]"]
+def is_property_with_options(key: str) -> bool:
+    """Check if a given property is known to be options-like requiring special parsing.
+
+    Args:
+        key: Property for which special options-like parsing decision has to be taken
+    """
+    return key in [
+        constants.OPTION_SPARK_DRIVER_DEFAULT_JAVA_OPTIONS,
+        constants.OPTION_SPARK_DRIVER_EXTRA_JAVA_OPTIONS,
+        constants.OPTION_SPARK_EXECUTOR_DEFAULT_JAVA_OPTIONS,
+        constants.OPTION_SPARK_EXECUTOR_EXTRA_JAVA_OPTIONS,
+    ]
```

Comment:
you can even create a constant that contains this list.

Reply:
Sure, will address this separately